src/validator.ts

```diff
@@ -1,19 +1,25 @@
 import { Natspec } from '../src/types/natspec.t';
+import { Config } from './utils';
 import { NodeToProcess } from './types/solc-typed-ast.t';
 import { EnumDefinition, ErrorDefinition, EventDefinition, FunctionDefinition, ModifierDefinition, StructDefinition, VariableDeclaration } from "solc-typed-ast";
 
-export function validate(node: NodeToProcess, natspec: Natspec): string[] {
-    let alerts: string[] = [];
+export function validate(node: NodeToProcess, natspec: Natspec, config: Config): string[] {
+    // There is inheritdoc, no other validation is needed
+    if(natspec.inheritdoc) return [];
+
+    // Inheritdoc is enforced but not present, returning an error
+    if(config.enforceInheritdoc && requiresInheritdoc(node)) return [`@inheritdoc is missing`];
 
-    if (natspec.inheritdoc) {
-        return alerts;
+    // Validate natspec for the constructor only if configured
+    if(node instanceof FunctionDefinition && node.kind === 'constructor') {
+        return config.constructorNatspec ? validateParameters(node, natspec) : [];
     }
 
-    if(!natspec.tags.length) {
-        alerts.push(`Natspec is missing`);
-        return alerts;
-    };
+    // Inheritdoc is not enforced nor present, and there is no other documentation, returning error
+    if(!natspec.tags.length && !natspec.inheritdoc) return [`Natspec is missing`];
```

Comment:
u have already checked if there is inheritdoc
```suggestion
    if(!natspec.tags.length) return [`Natspec is missing`];
```